processor/src/lib.rs

```diff
@@ -436,6 +446,14 @@ where
         Ok(())
     }
 
+    fn check_cycles_limit(&self) -> Result<(), ExecutionError> {
+        if self.max_cycles.is_some_and(|max_cycles| self.system.clk() > max_cycles) {
+            Err(ExecutionError::CycleLimitExceeded(self.max_cycles.unwrap()))
+        } else {
+            Ok(())
+        }
+    }
```

Comment:
We should probably perform this check in the [System::advance_clock()](https://github.com/0xPolygonMiden/miden-vm/blob/main/processor/src/system/mod.rs#L135) method so that we can fail immediately when clock value exceeds the max.

Also, I probably wouldn't create a separate function for this but just use a simple if statement.